### PR TITLE
identity cleanup

### PR DESCRIPTION
**Pull request by bryanchriswhite**
+ pluggable logging in key generation
+ remove old `SetupIdentity` and `SetupCA`

### CODE REVIEW

File: pkg/identity/certificate_authority.go [hunk 1]
```diff
@@ -97,16 +101,23 @@ func NewCA(ctx context.Context, opts NewCAOptions) (_ *FullCertificateAuthority,
...
 	fmt.Printf("Generating key with a minimum a difficulty of %d...\n", opts.Difficulty)
-	logStatus := func() {
-		count := atomic.LoadUint32(i)
-		hs := atomic.LoadUint32(highscore)
-		fmt.Printf("\rGenerated %d keys; best difficulty so far: %d", count, hs)
+	updateStatus := func() {
+		if opts.Logger != nil {
+			count := atomic.LoadUint32(i)
+			hs := atomic.LoadUint32(highscore)
+			_, err := opts.Logger.Write([]byte(fmt.Sprintf("\rGenerated %d keys; best difficulty so far: %d", count, hs)))
```
Comment (egonelbre) [inline pkg/identity/certificate_authority.go]: `fmt.Fprintf(opts.Logger, ...`

File: pkg/identity/certificate_authority.go [hunk 2]
```diff
@@ -97,16 +101,23 @@ func NewCA(ctx context.Context, opts NewCAOptions) (_ *FullCertificateAuthority,
...
+		}
 	}
 	err = GenerateKeys(ctx, minimumLoggableDifficulty, int(opts.Concurrency),
 		func(k *ecdsa.PrivateKey, id storj.NodeID) (done bool, err error) {
-			count := atomic.AddUint32(i, 1)
-			if count%100 == 0 {
-				logStatus()
+			if opts.Logger != nil {
+				count := atomic.AddUint32(i, 1)
+				if count%100 == 0 {
```
Comment (egonelbre) [inline pkg/identity/certificate_authority.go]: `if atomic.AddUint32(i, 1)%100 == 0 {`

File: pkg/identity/certificate_authority.go (hunk 1 above)
Comment (bryanchriswhite) [reply pkg/identity/certificate_authority.go]: nice - i didn't know about this

File: pkg/identity/certificate_authority.go (hunk 2 above)
Comment (bryanchriswhite) [reply pkg/identity/certificate_authority.go]: :man_facepalming:

**Pull request merged**